lib/elixir/src/elixir_expand.erl

```diff
@@ -972,6 +978,9 @@ format_error(wrong_number_of_args_for_super) ->
 format_error({invalid_arg_for_pin, Arg}) ->
   io_lib:format("invalid argument for unary operator ^, expected an existing variable, got: ^~ts",
                 ['Elixir.Macro':to_string(Arg)]);
+format_error({invalid_arg_for_lists_concatenation, Arg}) ->
+  io_lib:format("invalid argument for ++ operator, expected a list, got: ~ts",
```

Comment:
One last thing. We need to clarify that it is a literal list and that it is only during a match. What do you think about "invalid argument for ++ operator inside a match, expected a literal list, got: ~ts"?